### PR TITLE
add startup script test

### PR DESCRIPTION
**Pull request by gaohannk**
This is similar test like shutdown script. we only test startup script but not startup-script-url as it's already tested by wrapper.

### CODE REVIEW

File: imagetest/test_suites/metadata/setup.go [hunk 1]
```diff
@@ -1,17 +1,52 @@
...
 
 // Name is the name of the test package. It must match the directory name.
 var Name = "metadata"
 
+const (
+	startupScriptTemplate = `#!/bin/bash
+echo "%s" > %s`
+	startupOutputPath = "/startup_out.txt"
+	startupContent    = "The startup script worked."
+	startupMaxLength  = 32768 // max shutdown metadata value
```
Comment (EricEdens) [inline imagetest/test_suites/metadata/setup.go:18]: Can you add a link to where this number is defined? Is it a constant imposed by GCE?

File: imagetest/test_suites/metadata/startup_script_test.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	"testing"
+
+	"github.com/GoogleCloudPlatform/guest-test-infra/imagetest/utils"
+)
+
+// TestStartupScript test the standard metadata script.
+func TestStartupScript(t *testing.T) {
+	bytes, err := ioutil.ReadFile(startupOutputPath)
+	if err != nil {
+		t.Fatalf("failed to read shutdown output %v", err)
```
Comment (EricEdens) [inline imagetest/test_suites/metadata/startup_script_test.go:17]: do you mean startup output?

File: imagetest/test_suites/metadata/setup.go [hunk 3]
```diff
@@ -1,17 +1,52 @@
...
+	if err := vm2.Reboot(); err != nil {
+		return err
+	}
+	vm2.RunTests("TestStartupScript$")
+
+	vm3, err := t.CreateTestVM("vm3")
+	if err != nil {
+		return err
+	}
+	vm3.SetShutdownScript(strings.Repeat("a", startupMaxLength))
```
Comment (EricEdens) [inline imagetest/test_suites/metadata/setup.go:45]: I see that the test references "StartupScript", but this sets the shutdown script.

File: imagetest/test_suites/metadata/startup_script_test.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+import (
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/guest-test-infra/imagetest/utils"
+)
+
+// TestStartupScript test the standard metadata script.
```
Comment (EricEdens) [inline imagetest/test_suites/metadata/startup_script_test.go:13]: Can you add some more context to this?

File: imagetest/test_suites/metadata/startup_script_test.go [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+func TestStartupScript(t *testing.T) {
+	bytes, err := ioutil.ReadFile(startupOutputPath)
+	if err != nil {
+		t.Fatalf("failed to read shutdown output %v", err)
+	}
+	output := strings.TrimSpace(string(bytes))
+	if output != startupContent {
+		t.Fatalf(`shutdown script output expect "%s", but actually "%s"`, startupOutputPath, output)
```
Comment (EricEdens) [inline imagetest/test_suites/metadata/startup_script_test.go:21]: shutdown again

**Review by EricEdens — COMMENTED** [review]
Overall looks good! A few comments about adding comments for future readers, plus some notes regarding mixed usage of "startup" vs "shutdown".

File: imagetest/test_suites/metadata/setup.go (hunk 1 above)
Comment (gaohannk) [reply imagetest/test_suites/metadata/setup.go:18]: Yes. https://cloud.google.com/compute/docs/metadata/setting-custom-metadata#limitations

File: imagetest/test_suites/metadata/startup_script_test.go (hunk 2 above)
Comment (gaohannk) [reply imagetest/test_suites/metadata/startup_script_test.go:17]: Typo.

File: imagetest/test_suites/metadata/startup_script_test.go (hunk 5 above)
Comment (gaohannk) [reply imagetest/test_suites/metadata/startup_script_test.go:21]: Typo.

File: imagetest/test_suites/metadata/startup_script_test.go (hunk 4 above)
Comment (gaohannk) [reply imagetest/test_suites/metadata/startup_script_test.go:13]: TestStartupScript verify that the standard metadata script could run successfully by checking the output content

File: imagetest/test_suites/metadata/setup.go (hunk 3 above)
Comment (gaohannk) [reply imagetest/test_suites/metadata/setup.go:45]: Typo.

File: imagetest/test_suites/metadata/setup.go [hunk 6]
```diff
@@ -22,10 +26,12 @@ done`
 echo "%s" > %s`
 	shutdownOutputPath = "/shutdown_out.txt"
 	shutdownContent    = "The shutdown script worked."
-	shutdownMaxLength  = 32768 // max shutdown metadata value
+	// max metadata value https://cloud.google.com/compute/docs/metadata/setting-custom-metadata#limitations
+	metadataMaxLength = 32768
```
Comment (EricEdens) [inline imagetest/test_suites/metadata/setup.go:30]: Thanks for the link; I'm seeing "Each metadata value has a maximum limit of 256 KB" on that page -- is 32768 something else? Also, in the comment, can you specify the unit? something like: Maximum size of metadata value payload, in bytes.

Comment (gaohannk) [reply imagetest/test_suites/metadata/setup.go:30]: Interesting, i didn't pay much attention on this value. I found this link, it seems these is inconsistent which I think one is legacy value.
https://cloud.google.com/sdk/gcloud/reference/container/clusters/create

Comment (gaohannk) [reply imagetest/test_suites/metadata/setup.go:30]: ```Values must be less than or equal to 32,768 bytes in length. The total size of all keys and values must be less than 512 KB.```

**Review by EricEdens — APPROVED** [review]

**Pull request merged**